packages/core/src/parser/util.ts

```diff
@@ -506,6 +506,65 @@ export function stopBefore<N extends Returnable>(
 	}
 }
 
+/**
+ * @returns A parser that is based on the passed-in `parser`, but concatenates lines
+ * together when we reach, in order:
+ * - a backslash
+ * - whitespace (optional)
+ * - a newline
+ * - whitespace (optional)
+ */
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: InfallibleParser<N>,
+): InfallibleParser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N> {
+	return (src, ctx): Result<N> => {
+		let wrappedStr = src.sliceToCursor(0)
+		const wrapper = '\\'
+		const wrappedSrcCursor = wrappedStr.length
+		const indexMap: IndexMap = []
+
+		while (src.canRead()) {
+			wrappedStr += src.readUntil(wrapper)
+			if (!src.canRead()) {
+				// Stop wrapping `src` if we reach end of file or a terminator before a wrapper
```

Comment:
What does "or a terminator" mean here? What other option is there except reaching EOF?

Reply:
oops outdated comment